extension/src/pipeline/quickPick.ts

```diff
@@ -52,6 +52,69 @@ const formatFieldQuickPickValues = (
   return formattedFields
 }
 
+const verifyFilesHavingSingleField = (files: {
+  [file: string]: string[] | string
+}) => {
+  for (const [file, fields] of Object.entries(files)) {
+    if (Array.isArray(fields)) {
+      void Toast.showError(
+        `${file} cannot have more than one metric selected. See [an example](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#available-configuration-fields) of a plot with multiple x metrics.`
+      )
+      return
+    }
+  }
+
+  return files
+}
+
+const verifyXFields = (
+  xValues: { file: string; key: string }[] | undefined
+) => {
+  if (!xValues || xValues.length === 0) {
+    return
+  }
+
+  const x = formatFieldQuickPickValues(xValues)
+
+  const doFilesHaveOneField = verifyFilesHavingSingleField(x)
+
+  if (!doFilesHaveOneField) {
+    return
+  }
+
+  return { firstXKey: xValues[0].key, x: x as PlotConfigData['x'] }
+}
+
+const verifyYFields = (
+  yValues: { file: string; key: string }[] | undefined,
+  xFieldsLength: number
+) => {
+  if (!yValues || yValues.length === 0) {
+    return
+  }
+
+  const y = formatFieldQuickPickValues(yValues)
```

Comment:
There is a lot of repetition between `verifyXFields`and `verifyYFields` that could be extracted into a helper function.